gradle-baseline-java/src/main/groovy/com/palantir/baseline/extensions/BaselineErrorProneExtension.java

```diff
@@ -71,6 +71,7 @@
             // TODO(ckozak): re-enable pending scala check
             // "ThrowSpecificity",
             "UnsafeGaugeRegistration",
+            "PreferImmutableStreamExCollections",
```

Comment:
Limiting to streamex as opposed to standard streams reduces some danger, but I wouldn't be surprised if there are places which expect to mutate the result. I wouldn't want the robots to cause p0s.
It's unfortunate that the `toImmutable*()` methods don't return a special type like ImmutableList where the type system could help us out. Flow analysis is more difficult.

Reply:
Yep, this is a theoretical possibility. In practise, I think it's gonna be a very small number of repos who rely on this kind of mutability and they'd _also_ need to have no unit/integration test coverage in order for a release to get out.  Moreover we have blue-green deploys which are _supposed_ to minimize the user-facing impact if something slips through the net 🤷

Reply:
What if we don't auto-fix, but update the description to include both a warning about mutability, and instructions to apply the fixes `./gradlew classes -PerrorProneApply=PreferImmutableStreamExCollections
`

Reply:
I think I wanna stick with the autofix here. Immutability is such a core tenet of conjure and the way we write software at Palantir, that I think it's worth ratcheting this towards a better world. This PR should have a pretty minimal scope, because it won't touch the regular usage of java.util.streams.Stream.

Also since this is potentially gonna make code changes, people will have to +1 it before it goes in.